# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1204276143 18000
#      Fri Feb 29 04:09:03 2008 -0500
# Node ID b84c5cbc081243fabec4026636f0c947c587c372
# Parent  b1ff001022af99d187832d01222c48c0b0a386dd
print.m: handle gif and jpg devices

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-02-29  John W. Eaton  <jwe@octave.org>
+
+	* plot/print.m: Handle gif and jpg devices.
+
 2008-02-28  John W. Eaton  <jwe@octave.org>
 
 	* plot/__go_draw_axes__.m (get_old_gnuplot_color): New subfunction.
 	Use it to replace repeated if/elseif blocks.
 
 2008-02-28  Kai Habel  <kai.habel@gmx.de>
 
 	* plot/__go_draw_axes__.m: If facecolor property of a surface is
diff --git a/scripts/plot/print.m b/scripts/plot/print.m
--- a/scripts/plot/print.m
+++ b/scripts/plot/print.m
@@ -77,16 +77,21 @@
 ##     whether the special flag should be set for the text in the figure
 ##     (default is @code{-textnormal}). 
 ##   @item hpgl
 ##     HP plotter language
 ##   @item mf
 ##     Metafont
 ##   @item png
 ##     Portable network graphics
+##   @item jpg
+##   @itemx jpeg
+##     JPEG image
+##   @item gif
+##     GIF image
 ##   @item pbm
 ##     PBMplus
 ##   @item svg
 ##     Scalable vector graphics
 ##   @item pdf
 ##     Portable document format
 ##   @end table
 ##
@@ -213,20 +218,21 @@ function print (varargin)
     endif
   elseif (strcmp (dev, "ill"))
     dev = "aifm";
   elseif (strcmp (dev, "cdr"))
     dev = "corel";
   endif
 
   ## check if we have to use convert
-  dev_list = {"aifm", "corel", "fig", "png", "pbm", "dxf", "mf", "svg", ...
-	      "hpgl", "ps", "ps2", "psc", "psc2", "eps", "eps2", ...
-	      "epsc", "epsc2", "emf", "pstex", "pslatex", ...
-	      "epslatex", "epslatexstandalone", "pdf"};
+  dev_list = {"aifm", "corel", "fig", "png", "jpg", "jpeg", ...
+	      "gif", "pbm", "dxf", "mf", "svg", "hpgl", ...
+	      "ps", "ps2", "psc", "psc2", "eps", "eps2", ...
+	      "epsc", "epsc2", "emf", "pdf", "pslatex", ...
+	      "epslatex", "epslatexstandalone", "pstex"};
   convertname = "";
   [idx, errmsg] = cellidx (dev_list, dev);
   if (! idx)
     if (! isempty (devopt))
       convertname = cstrcat (devopt, ":", name);
     else
       convertname = name;
     endif
@@ -332,17 +338,22 @@ function print (varargin)
       options = cstrcat (options, " \"", font, "\"");
     endif
     if (! isempty (fontsize))
       options = cstrcat (options, " ", fontsize);
     endif
 
     new_terminal = cstrcat ("emf ", options);
 
-  elseif (strcmp (dev, "png") || strcmp (dev, "pbm"))
+  elseif (strcmp (dev, "png") || strcmp (dev, "gif")
+	  || strcmp (dev, "jpg") || strcmp (dev, "jpeg")
+	  || strcmp (dev, "pbm"))
+    if (strcmp (dev, "jpg"))
+      dev = "jpeg";
+    endif
     ## Portable network graphics, PBMplus
 
     ## FIXME -- New PNG interface takes color as "xRRGGBB"
     ## where x is the literal character 'x' and 'RRGGBB' are the red,
     ## green and blue components in hex.  For now we just ignore it
     ## and use default.  The png terminal now is so rich with options,
     ## that one perhaps has to write a separate printpng.m function.
     ## DAS
diff --git a/src/xpow.cc b/src/xpow.cc
--- a/src/xpow.cc
+++ b/src/xpow.cc
@@ -68,24 +68,28 @@ xisint (double x)
 //                  +---+---+----+----+
 //   complex_matrix | 4 | * | 10 |  * |
 //                  +---+---+----+----+
 
 // -*- 1 -*-
 octave_value
 xpow (double a, double b)
 {
+  double retval;
+
   if (a < 0.0 && static_cast<int> (b) != b)
     {
       Complex atmp (a);
 
       return std::pow (atmp, b);
     }
   else
-    return std::pow (a, b);
+    retval = std::pow (a, b);
+
+  return retval;
 }
 
 // -*- 2 -*-
 octave_value
 xpow (double a, const Matrix& b)
 {
   octave_value retval;
 
